File: PS Multi Tools/Classes/TelnetClient.vb
Imports System.Net.Sockets
Imports System.Text
Imports System.Threading

Public Class TelnetClient

    Private TCPSocket As TcpClient

    Private NewNetworkStream As NetworkStream

    Public Sub New(HostName As String, Port As Integer)
        TCPSocket = New TcpClient()
        TCPSocket.Connect(HostName, Port)
        NewNetworkStream = TCPSocket.GetStream()
    End Sub

    Public Sub Write(Command As String)
        If NewNetworkStream Is Nothing Then Return

        Dim CommandWithNewLine As String = Command & vbCrLf
        Dim CommandBytes As Byte() = Encoding.ASCII.GetBytes(CommandWithNewLine)
        NewNetworkStream.Write(commandBytes, 0, commandBytes.Length)
        NewNetworkStream.Flush()
    End Sub

    Public Function Read() As String
        If NewNetworkStream Is Nothing Then Return String.Empty

        Dim ResponseBuilder As New StringBuilder()
        Dim Buffer(1024) As Byte

        While NewNetworkStream.DataAvailable
            Dim BytesRead As Integer = NewNetworkStream.Read(Buffer, 0, Buffer.Length)
            If bytesRead <= 0 Then Exit While
            responseBuilder.Append(Encoding.ASCII.GetString(buffer, 0, bytesRead))
            Thread.Sleep(50)
        End While

        Return responseBuilder.ToString()
    End Function

    Public Sub Close()
        If NewNetworkStream IsNot Nothing Then NewNetworkStream.Close()
        If TCPSocket IsNot Nothing Then TCPSocket.Close()
    End Sub

End Class

File: PSMultiTools/Classes/TelnetClient.cs
using System.Net.Sockets;
using System.Text;
using System.Threading;

namespace PSMultiTools.Classes
{

    public class TelnetClient
    {

        public TcpClient TCPSocket;
        public NetworkStream NewNetworkStream;

        public TelnetClient(string HostName, int Port)
        {
            TCPSocket = new TcpClient();
            TCPSocket.Connect(HostName, Port);
            NewNetworkStream = TCPSocket.GetStream();
        }

        public void Write(string Command)
        {
            if (NewNetworkStream is null)
                return;

            string CommandWithNewLine = Command + "\r\n";
            byte[] CommandBytes = Encoding.ASCII.GetBytes(CommandWithNewLine);
            NewNetworkStream.Write(CommandBytes, 0, CommandBytes.Length);
            NewNetworkStream.Flush();
        }

        public string Read()
        {
            if (NewNetworkStream is null)
                return string.Empty;

            var ResponseBuilder = new StringBuilder();
            var Buffer = new byte[1025];

            while (NewNetworkStream.DataAvailable)
            {
                int BytesRead = NewNetworkStream.Read(Buffer, 0, Buffer.Length);
                if (BytesRead <= 0)
                    break;
                ResponseBuilder.Append(Encoding.ASCII.GetString(Buffer, 0, BytesRead));
                Thread.Sleep(50);
            }

            return ResponseBuilder.ToString();
        }

        public void Close()
        {
            NewNetworkStream?.Close();
            TCPSocket?.Close();
        }

    }
}